lib/CodeGen/CGBuiltin.cpp

```diff
@@ -2426,6 +2426,9 @@ RValue CodeGenFunction::EmitBuiltinExpr(const FunctionDecl *FD,
     break;
   }
   case Builtin::BI_Dynamic_check: {
+    if (!getLangOpts().CheckedC)
+      break;
```

Comment:
Can this ever happen?  We should (1) not recognize _Dynamic_check is an intrinsic if Checked C is turned off and (2) have an llvm_unreachable statement before the break, to trigger an internal compiler error if we get here.